pkg/controller/migmigration/task.go

```diff
@@ -525,9 +515,6 @@ func (t *Task) next() {
 		if next.flags&HasPVs != 0 && !t.hasPVs() {
 			continue
 		}
-		if next.flags&HasStagePods != 0 && !t.Owner.Status.HasCondition(StagePodsCreated) {
```

Comment:
The `HasStagePods` flag is still being used.  Two choices here:
1. Set the condition (perhaps renamed to `HasStagePods`) when at least 1 application pod is annotated. Continues to provide useful information to the user and supports skipping the stage pod delete/terminate phases.
2. Get rid of the flag.  Simpler.